src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -145,16 +145,22 @@ MultiValue GetTypeValueNodeFromGenericArgument (TypeReference genericArgument)
 				// Technically this should be a new value node type as it's not a System.Type instance representation, but just the generic parameter
 				// That said we only use it to perform the dynamically accessed members checks and for that purpose treating it as System.Type is perfectly valid.
 				return new GenericParameterValue (inputGenericParameter, _context.Annotations.FlowAnnotations.GetGenericParameterAnnotation (inputGenericParameter));
-			} else {
-				TypeDefinition? genericArgumentTypeDef = ResolveToTypeDefinition (genericArgument);
-				if (genericArgumentTypeDef != null) {
-					return new SystemTypeValue (genericArgumentTypeDef);
-				} else {
-					// If we can't resolve the generic argument, it means we can't apply potential requirements on it
-					// so track it as unknown value. If we later on hit this unknown value as being used somewhere
-					// where we need to apply requirements on it, it will generate a warning.
-					return UnknownValue.Instance;
+			} else if (ResolveToTypeDefinition (genericArgument) is TypeDefinition genericArgumentType) {
+				if (genericArgumentType.IsTypeOf ("System", "Nullable`1")) {
+					var innerGenericArgument = (genericArgument as IGenericInstance)?.GenericArguments.FirstOrDefault ();
+					switch (innerGenericArgument) {
+					case GenericParameter gp:
+						return new NullableValueWithDynamicallyAccessedMembers (genericArgumentType,
+							new GenericParameterValue (gp, _context.Annotations.FlowAnnotations.GetGenericParameterAnnotation (gp)));
+
+					case TypeReference underlyingType:
+						return new NullableSystemTypeValue (genericArgumentType, ResolveToTypeDefinition (underlyingType)!);
```

Comment:
Probably uncommon, but I think it's possible for `ResolveToTypeDefinition` to fail - maybe this should return an `UnknownValue` if it resolves to null?